qfile.c

```diff
@@ -57,11 +60,77 @@ struct qfile_opt_state {
 	bool slotted;
 	bool basename;
 	bool exact;
+	bool skip_plibreg;
 	bool orphans;
 	bool assume_root_prefix;
 };
 
 /*
+ * As a final step, check if file is in the plib_reg
+ */
+static int qfile_check_plibreg(void *priv)
+{
+	struct qfile_opt_state *state = priv;
+
+	int fd_plibreg;
+        FILE *fp_plibreg;
+	struct stat cst;
+
+	/* Open plibreg */
+        fp_plibreg = NULL;
+	fd_plibreg = open("/var/lib/portage/preserved_libs_registry", O_RDONLY|O_CLOEXEC, 0);
+	if (fd_plibreg == -1)
+		return 0;
+	if (fstat(fd_plibreg, &cst)) {
+		close(fd_plibreg);
+		return 0;
+	}
+	if ((fp_plibreg = fdopen(fd_plibreg, "r")) == NULL) {
+		close(fd_plibreg);
+		return 0;
+	}
+
+        qfile_args_t *args = &state->args;
+        char **base_names = args->basenames;
+        char **dir_names = args->dirnames;
+        short *non_orphans = args->non_orphans;
+        int *results = args->results;
+	char file[_Q_PATH_MAX];
+	char *line = NULL;
+	size_t len = 0;
+	int found = 0;
+
+	for (int i = 0; i < args->length; i++) {
+		if (base_names[i] == NULL)
+			continue;
+		if (non_orphans && non_orphans[i])
+			continue;
+		if (results[i] == 1)
+			continue;
+
+		snprintf(file, sizeof(file), "%s/%s", dir_names[i], base_names[i]);
+
+		while (getline(&line, &len, fp_plibreg) != -1)
+			if (strstr(line, file) != NULL) {
```

Comment:
Is this going to respect the -b flag?

Reply:
Thx, no, it didn't. Fixed by only putting dir_names[i] in there if it isn't empty.

Reply:
honestly I don't know what I was talking about here, there is no such thing as a `-b` flag in today's qfile.

Reply:
-b used to be needed if only basename was provided, this became default behaviour after 5531d5c
from the looks of it this change is needed, but I can't test as my system is on -0.74

Reply:
On -0.74 the change makes -b work though :D